// node-yaml-config/dist/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.reloadAsync = exports.loadAsync = exports.readAsync = exports.reload = exports.load = exports.read = void 0;
const config_loader_1 = require("./config-loader");
__exportStar(require("./config-loader"), exports);
__exportStar(require("./config-loader.options"), exports);
exports.read = config_loader_1.DEFAULT_CONFIG_LOADER.read.bind(config_loader_1.DEFAULT_CONFIG_LOADER);
exports.load = config_loader_1.DEFAULT_CONFIG_LOADER.load.bind(config_loader_1.DEFAULT_CONFIG_LOADER);
exports.reload = config_loader_1.DEFAULT_CONFIG_LOADER.reload.bind(config_loader_1.DEFAULT_CONFIG_LOADER);
exports.readAsync = config_loader_1.DEFAULT_CONFIG_LOADER.readAsync.bind(config_loader_1.DEFAULT_CONFIG_LOADER);
exports.loadAsync = config_loader_1.DEFAULT_CONFIG_LOADER.loadAsync.bind(config_loader_1.DEFAULT_CONFIG_LOADER);
exports.reloadAsync = config_loader_1.DEFAULT_CONFIG_LOADER.reloadAsync.bind(config_loader_1.DEFAULT_CONFIG_LOADER);
